packages/zod/src/index.ts

```diff
@@ -224,15 +229,15 @@ export const generateZodValidationSchemaDefinition = (
       if (schema.additionalProperties) {
         functions.push([
           'additionalProperties',
-          isBoolean(schema.additionalProperties)
-            ? schema.additionalProperties
-            : generateZodValidationSchemaDefinition(
-                schema.additionalProperties as SchemaObject,
-                context,
-                true,
-                name,
-                strict,
-              ),
+          generateZodValidationSchemaDefinition(
+            isBoolean(schema.additionalProperties)
```

Comment:
Actual fix which I'm not sure about: since `additionalProperties: true` is just and alias for `additionalProperties: {}`, I decided to treat it as latter. Should definition be faithful to original scheme?

Reply:
Is it correct to understand that if it is simply `true`, it is determined that the additional property does not exist?

Reply:
As far as I understood from the OpenAPI docs, it's the opposite: 

> If the dictionary values can be of any type (aka free-form object), use additionalProperties: true:

https://swagger.io/docs/specification/data-models/dictionaries/#free-form

So, `additionalProperties: true` is equivalent to `additionalProperties: {}`, which means that any type of property values allowed. Basically, it means it's `Record<string, any>`. `additionalProperties: {}` works fine, btw, it's just `additionalProperties: true` that's broken.

Reply:
Thanks, I understood.